search-service/src/main/kotlin/com/egm/stellio/search/entity/web/EntityHandler.kt

```diff
@@ -102,16 +101,17 @@ class EntityHandler(
             } else BatchOperationResult() to expandedEntity
 
         if (remainingEntity != null) {
-            either {
-                val ngsiLdEntity = remainingEntity.toNgsiLdEntity().bind()
-                entityService.createEntity(ngsiLdEntity, remainingEntity, sub.getOrNull()).bind()
-            }.fold(
-                { result.errors.add(BatchEntityError(entityId, it.toProblemDetail())) },
-                { result.success.add(BatchEntitySuccess(NGSILD_LOCAL)) }
+            result.addEither(
+                either {
+                    val ngsiLdEntity = remainingEntity.toNgsiLdEntity().bind()
+                    entityService.createEntity(ngsiLdEntity, remainingEntity, sub.getOrNull()).bind()
+                },
+                entityId,
+                null
             )
         }
 
-        result.toNonBatchEndpointResponse(entityId)
+        result.toNonBatchEndpointResponse(entityId, successStatus = HttpStatus.CREATED, addLocation = true)
```

Comment:
not sure I like to name all the parameters. I agree for the boolean `addLocation`. I find it is redundant for the HTTP status (and entity id in the delete handler)